R/bainCommon.R

```diff
@@ -0,0 +1,1288 @@
+  
+  set.seed(options[["seed"]])
+  
+  if(type %in% c("onesampleTTest", "independentTTest")){
+    result <- .bainGetTTestResults(dataset, options, bainContainer, ready, type, variable, testType)
+  } else if(type == "pairedTTest"){
+    result <- .bainGetPairedTTestResults(dataset, options, bainContainer, ready, type, pair, testType) 
+  } else if (type %in% c("anova", "ancova", "regression", "sem")){
+    result <- .bainGetRegressionResults(dataset, options, bainContainer, ready, type)
+  }
+  
+  return(result) 
+}
+
+.bainGetTTestResults <- function(dataset, options, bainContainer, ready, type, variable, testType){
+  
+  if(!is.null(bainContainer[[variable]]))
+    return(bainContainer[[variable]]$object)
+  
+  variableData <- dataset[[variable]]
+  testValue <- format(options[["testValue"]], scientific = FALSE)  
+  
+  p <- try({
+    if(type == "onesampleTTest"){
+      bainResult <- .bainTTestRaw(options, x = variableData, nu = testValue, type = testType)
+    } else if(type == "independentTTest"){
+      levels <- base::levels(dataset[[options[["groupingVariable"]]]])
+      if (length(levels) != 2) {
+        g1 <- "1"
+        g2 <- "2"
+      } else {
+        g1 <- levels[1]
+        g2 <- levels[2]
+      } 
+      subDataSet <- dataset[, c(variable, options[["groupingVariable"]])]
+      group1 <- subDataSet[subDataSet[[options[["groupingVariable"]]]]== g1, variable]
+      group2 <- subDataSet[subDataSet[[options[["groupingVariable"]]]]== g2, variable]
+      bainResult <- .bainTTestRaw(options, x = group1, y = group2, type = testType)
+    }
+  })
+  
+  if (isTryError(p)) {
+    bainContainer$setError(gettextf("An error occurred in the analysis:<br>%1$s.", jaspBase::.extractErrorMessage(p)))
+    return()
+  }
+  
+  bainContainer[[variable]] <- createJaspState(bainResult, dependencies = c("testValue", "hypothesis"))
+  bainContainer[[variable]]$dependOn(optionContainsValue = list("variables" = variable))
+  return(bainContainer[[variable]]$object)
+}
+
+.bainGetPairedTTestResults <- function(dataset, options, bainContainer, ready, type, pair, testType){
+  
+  currentPair <- paste(pair, collapse = " - ")
+  
+  if(!is.null(bainContainer[[currentPair]]))
+    return(bainContainer[[currentPair]]$object)
+  
+  if (pair[[2]] != "" && pair[[1]] != pair[[2]] && pair[[1]] != "") {
+    
+    subDataSet <- subset(dataset, select=c(pair[[1]], pair[[2]]) )
+    c1 <- subDataSet[[ pair[[1]] ]]
+    c2 <- subDataSet[[ pair[[2]] ]]  
+    
+    p <- try({
+      bainResult <- .bainTTestRaw(options, x = c1, y = c2, type = testType, paired = TRUE)
+    })
+    
+    if(isTryError(p)){
+      bainContainer$setError(gettextf("An error occurred in the analysis:<br>%1$s.", jaspBase::.extractErrorMessage(p)))
+      return()
+    }
+    
+    bainContainer[[currentPair]] <- createJaspState(bainResult, dependencies = "hypothesis")
+    bainContainer[[currentPair]]$dependOn(optionContainsValue = list("pairs" = pair))
```

Comment:
I wonder if this works, see also https://github.com/jasp-stats/jaspTTests/pull/35

Reply:
As long as `pair` is a character vector, I think it works fine.

Reply:
I also think so actualy

Reply:
@vandenman: just for my understanding and to verify the correctness of the ttest changes:

If you have the following paired variables input:
[ VarA - VarB ]
[ VarB - VarC ]
[ VarA - VarC ]

Would the third pair disappear from the results if removed from the variables input?
Considering VarA and VarC are technically still present.

Reply:
I can show:

![image](https://user-images.githubusercontent.com/25059399/108053353-95072c00-704d-11eb-88ba-529953e40c92.png)

Removing the third pair removes the last row from the table:
![image](https://user-images.githubusercontent.com/25059399/108053404-a819fc00-704d-11eb-8977-d84e2b443f8a.png)